invenio_rdm_records/services/access/service.py

```diff
@@ -749,3 +753,40 @@ def create_guest_access_request(self, identity, token, expand=False, uow=None):
             data=comment,
             uow=uow,
         )
+
+    @unit_of_work()
+    def update_access_settings(
+        self,
+        identity,
+        id_,
+        data,
+        uow=None,
+    ):
+        """Update access settings for a record (resp. its parent)."""
+        record, parent = self.get_parent_and_record_or_draft(id_)
+
+        # Permissions
+        self.require_permission(identity, "manage", record=record)
+
+        # Validation
+        data, __ = self.schema_access_settings.load(
+            data, context=dict(identity=identity), raise_errors=True
+        )
+
+        # Update
+        setattr(parent.access, "settings", data)
+
+        # Commit
+        uow.register(RecordCommitOp(parent))
+        if record:
```

Comment:
from @ntarocco https://github.com/inveniosoftware/invenio-rdm-records/pull/1392/files/51716d8ab61b74ddaf667b6849a2467436bdcee1#r1280814271 . Do we dump the access settings in the record? Can you check if a record stores in the json the access information?